technic/machines/HV/quarry.lua

```diff
@@ -112,6 +120,15 @@ local function quarry_receive_fields(pos, formname, fields, sender)
 			reset_quarry(pos)
 		end
 	end
+
+	if fields.max_depth then
+		-- apply max-depth config
+		local max_depth = tonumber(fields.max_depth)
+		if max_depth and max_depth > 0 and max_depth <= quarry_max_depth then
+			meta:set_int("max_depth", max_depth)
+		end
```

Comment:
I think it would be better to do same thing here, would be good for ux to set maximum possible value if larger than max is given as input.